### DISCUSSION TITLE
Render on VSCode remote login

### DISCUSSION THREAD

**Discussion by vmmhep** [quarto-dev/quarto-cli#2614] [Q&A] [answered]
Hi!
Is there a way to render quarto files locally via the VSCode extension if I'm connected via ssh? i.e. my .qmd files are on a server, but I cannot install quarto there.

Sorry as I'm sure this has been asked before but I cannot find it anywhere :(

Thanks in advance for any pointers!

**Comment by mcanouil** [comment, marked as answer]
Hum, this is not how remote development over SSH via VSCode works (or any system actually).
You can render the document or whatever really, then in the sidebar, explorer tab, click right for the contextual menu and then press the download menu to download locally the rendered document.

You can do the other way around, do everything locally, then connect over SSH and upload (drag and drop) the rendered file.

Edit: far as I can tell, the `quarto publish` command can't be used with custom ssh/sftp/etc. servers/protocols, thus after `quarto render`, you need to implement the upload to the server part yourself using whatever file transfer protocol suited for your server.

**Reply by vmmhep** [reply]
> Hum, this is not how remote development over SSH via VSCode works (or any system actually). You can render the document or whatever really, then in the sidebar, explorer tab, click right for the contextual menu and then press the download menu to download locally the rendered document.

I guess this involves installing `quarto` on the cluster I'm ssh-ing into?

> You can do the other way around, do everything locally, then connect over SSH and upload (drag and drop) the rendered file.
> 
> Edit: far as I can tell, the `quarto publish` command can't be used with custom ssh/sftp/etc. servers/protocols, thus after `quarto render`, you need to implement the upload to the server part yourself using whatever file transfer protocol suited for your server.

The main reason I wanted to do this is to do some data analytics (the server has the data on).  I've tried `pluto` but I wanted to give `quarto` a shot too... But if it's unsuitable then that's just how it is :shrug:

Thanks for getting back to me! :)

**Reply by eitsupi** [reply]
> I guess this involves installing `quarto` on the cluster I'm ssh-ing into?

Yes. Please check https://code.visualstudio.com/docs/remote/ssh

**Reply by vmmhep** [reply]
> Yes. Please check https://code.visualstudio.com/docs/remote/ssh

I checked this and can now run `quarto` on the cluster.  However, I get this error:
`version 'GLIBC_2.18' not found`
As far as I can tell from the link above, `glibc >= 2.17` is required (which the cluster has).  Anyone know if there's an older version of `quarto` I can use that only needs `glibc 2.17`?  Or is it VSCode that needs downgrading?
Thanks!
